src/plasma/plasma_store.cc

```diff
@@ -340,33 +239,37 @@ void return_from_get(PlasmaStoreState *store_state, GetRequest *get_req) {
    * tables if it is present there. It should only be present there if the get
    * request timed out. */
   for (ObjectID &object_id : get_req->object_ids) {
-    remove_get_request_for_object(store_state, object_id, get_req);
+    auto &get_requests = object_get_requests_[object_id];
+    /* Erase get_req from vector, see http://stackoverflow.com/a/16013546 */
```

Comment:
Hmm.. I just tried out my suggestion and it causes a crash in the plasma tests. It's probably not quite right.